### PR TITLE
Jbtm 3817 Store initialisation code does not report the reason for failures

### PR DESCRIPTION
**Pull request by mmusgrov**
https://issues.redhat.com/browse/JBTM-3817

This PR reports the reason for store initialisation failures.

CORE !TOMCAT !AS_TESTS !RTS !JACOCO !XTS !QA_JTA !QA_JTS_OPENJDKORB !PERFORMANCE !LRA !DB_TESTS !mysql !db2 !postgres !oracle

### CODE REVIEW

File: common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java [hunk 1]
```diff
@@ -23,7 +23,7 @@ public class ClassloadingUtility
 {
     /**
      * Load a class. No instantiation.
-     * 
+     * <p>
```
Comment (jmfinelli) [inline common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java:26]: This (and the others) should go in a different commit (dedicated to a change of style)

Comment (jmfinelli) [reply common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java:26]: and...btw...shouldn't it be confined with a `<\p>`. Just asking for curiosity

File: narayana-bom-test/pom.xml [hunk 2]
```diff
@@ -17,7 +17,7 @@
   <name>Narayana Bom Test</name>
   <description>Narayana BOM is for centralising dependency management for development purposes (test scoped dependencies)</description>
 
-<!--
+  <!--
```
Comment (jmfinelli) [inline narayana-bom-test/pom.xml:20]: Also this one should go to a dedicated commit for style changes

File: narayana-bom-test/pom.xml [hunk 3]
```diff
@@ -17,7 +17,7 @@
...
 	This Bill of Materials (BOM) is for centralising dependency management across the various modules of Narayana.
     The BOM is intended for development purposes within Narayana.
 -->
```
Comment (jmfinelli) [inline narayana-bom-test/pom.xml:23]: shouldn't we add spaces here as well?

File: ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/objectstore/StoreManager.java [hunk 4]
```diff
@@ -129,16 +129,8 @@ private static final ObjectStoreAPI initStore(String name)
...
-
-        try
-        {
-            store = ClassloadingUtility.loadAndInstantiateClass(ObjectStoreAPI.class, storeType, name);
-        }
-        catch (final Throwable ex)
-        {
-            throw new FatalError(tsLogger.i18NLogger.get_StoreManager_invalidtype() + " " + storeType, ex);
-        }
+        ObjectStoreAPI store = ClassloadingUtility.loadAndInstantiateClass(ObjectStoreAPI.class, storeType, name,
```
Comment (jmfinelli) [inline ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/objectstore/StoreManager.java:132]: Do we need a new line here?

File: common/classes/com/arjuna/common/logging/commonI18NLogger.java [hunk 5]
```diff
@@ -51,17 +52,18 @@ public interface commonI18NLogger {
...
-    public void warn_common_ClassloadingUtility_5(String arg0, @Cause() Throwable arg1);
+    public String warn_common_ClassloadingUtility_5(String arg0, @Cause() Throwable arg1);
 
     @Message(id = 48008, value = "cannot initialize from string {0}", format = MESSAGE_FORMAT)
     @LogMessage(level = WARN)
     public void warn_common_ClassloadingUtility_6(String arg0, @Cause() Throwable arg1);
 
+    @Message(id = 48009, value = "cannot create new instance of {0}", format = MESSAGE_FORMAT)
+    @LogMessage(level = FATAL)
+    public void fatal_common_ClassloadingUtility_7(String arg0, @Cause() Throwable arg1);
```
Comment (jmfinelli) [inline common/classes/com/arjuna/common/logging/commonI18NLogger.java:66]: Where do you use this FATAL log message?

**Review by jmfinelli — CHANGES_REQUESTED** [review]
Just minor changes

File: common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java (hunk 1 above)
Comment (mmusgrov) [reply common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java:26]: Oh okay then, bit of a nit pic but I'll remove them.
Note that empty elements do not require an explicit closing tag so I'll use `<p/>`.

Comment (mmusgrov) [reply common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java:26]: But something for others to note is that this is what Intellij added.

File: narayana-bom-test/pom.xml (hunk 2 above)
Comment (mmusgrov) [reply narayana-bom-test/pom.xml:20]: Strange, I don't recall making this change, I'll back it out.

File: common/classes/com/arjuna/common/logging/commonI18NLogger.java (hunk 5 above)
Comment (mmusgrov) [reply common/classes/com/arjuna/common/logging/commonI18NLogger.java:66]: I don't so I will remove it.

File: narayana-bom-test/pom.xml (hunk 3 above)
Comment (mmusgrov) [reply narayana-bom-test/pom.xml:23]: I don't know how this one ended up in the commit, I'll remove it.

File: common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java (hunk 1 above)
Comment (jmfinelli) [reply common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java:26]: Good to know :-)

File: narayana-bom-test/pom.xml (hunk 3 above)
Comment (mmusgrov) [reply narayana-bom-test/pom.xml:23]: Found it, by default we [enable sortpom](https://github.com/jbosstm/narayana/blob/7.0.0.Final/pom.xml#L178). Is that a bug or a feature (I'm sure we've discussed it before but any new pom should verify that it passes sort pom so we all need to be vigilant to check that during reviews).

**Comment by mmusgrov** [comment]
@jmfinelli all of the conversations are showing as resolved before I pushed an update for them. Did you resolve them all?

**Comment by marcosgopen** [comment]
Good catch @mmusgrov !

**Comment by jmfinelli** [comment]
> @jmfinelli all of the conversations are showing as resolved before I pushed an update for them. Did you resolve them all?

Yep :-)

**Review by jmfinelli — APPROVED** [review]

**Comment by mmusgrov** [comment]
Thanks for the review Manuel and catching a couple of things that I missed.
